speech/tests/system.py

```diff
@@ -245,3 +247,59 @@ def test_stream_recognize_single_utterance(self):
             for results in self._make_streaming_request(
                     file_obj, single_utterance=False):
                 self._check_results(results.alternatives)
+
+    def test_long_running_recognize(self):
+
+        def _build_large_audio_file(wav, reps):
+            ''' From https://stackoverflow.com/questions/46538867/
+                             attributeerror-wave-write-instance-has-no-
+                             attribute-exit
+            '''
+            def __enter__(self):
+                return self
+            def __exit__(self, exc_type, exc_value, traceback):
+                self.close()
+
+            wave.Wave_read.__exit__ = wave.Wave_write.__exit__ = __exit__
+            wave.Wave_read.__enter__ = wave.Wave_write.__enter__ = __enter__
+            data = []
+
+            ''' From https://stackoverflow.com/questions/2890703/
+                             how-to-join-two-wav-files-using-python
+            '''
+            for rep in range(0, reps):
```

Comment:
There is no need to loop over opening the file:  just open it once and read the values (note that I open the stream explicitly first, to ensure the close semantics you are aiming for above.

```python
        with open(AUDIO_FILE, 'rb') as in_stream:
            with wave.open(in_stream) as in_wavfile:
                params = in_wavfile.getparams()
                frames = in_wavfile.readframes(wavefile.getnframes())
```

For the temporary, you could just use an `io.BytesIO` instance, and then use `params` and `frames` to construct the big wav

```python

        with io.BytesIO() as out_stream:
            with wave.open(out_stream, 'w') as out_wavfile:
                out_wavfile.setparams(params)
                for rep in range(15):
                    out_wavfile.writeframes(frames)
            content = out_stream.getvalue()
```

Reply:
so I'm getting a problem that if I mock and patch it I'm not getting a large audio file, I'll look at it again later.